app/models/location_request.rb

```diff
@@ -0,0 +1,7 @@
+class LocationRequest
+  include ActiveModel::Model
+
+  attr_accessor :postcode
+
+  validates :postcode, postcode: true
```

Comment:
You could do something here like:

```
class PostcodeChecker
  include ActiveModel::Model

  attr_accessor :postcode

  validates :postcode, postcode: true

  def self.valid?(args)
    new(postcode: args).valid?
  end
end
```

And then in the controller:

```
def validate_postcode
  unless PostcodeChecker.valid?(params[:postcode])
     render json: { errors: @req.errors }, status: 400
  end
end
```

Reply:
I have refactored it, thanks @hannako !